### ISSUE TITLE
index row size  exceeds maximum 2712 for index "tx_data_index" 

### ISSUE THREAD

**Issue by ramilexe** [vulcanize/ipld-eth-indexer#61]
When run syncing with testnet sometimes appear error message in logs.

Errors from logs when run on testnet
```
ERRO[2020-10-16T16:50:54+03:00] ethereum sync worker 1 transformer error: pq: index row size 3488 exceeds maximum 2712 for index "tx_data_index" 
```

Row to reproduce:
```
# run statediff geth (testnet)
./geth-statediff --testnet --rpc --rpcapi "web3,eth,net,debug,personal,statediff" --rpccorsdomain '*' --rpcaddr="0.0.0.0" --rpcport="8545" --syncmode=full --statediff --ws --wsaddr="0.0.0.0"

# run ipld-eth-indexer 
ETH_WS_PATH=localhost:8546 ./ipld-eth-indexer sync --database-password password --database-user vdbm --database-port 5420 --eth-chain-id 3
```

**Comment by i-norden** [comment]
Will look into this

**Comment by i-norden** [comment]
As the error suggests, the tx `data` row can be too large to be btree indexed

Quick fix: DROP the `tx_data_index`

Longer-term: Could create a new column to hold the hash of the keccak `data`, and index this instead. A stored function or external program could direct look ups on `data` through its hash. This needs to take into consideration the cost of hashing.

**Comment by i-norden** [comment]
Fixed in #66 and in the v0.5.1-alpha release (which works with currently running version of statediffing geth (v1.9.11-statediff-0.0.5)

Opening new issue for longer term option of indirect indexing of tx data by a hash column
